code/modules/unit_tests/~bubber/ensure_pref_sanity.dm

```diff
@@ -0,0 +1,52 @@
+/datum/unit_test/bubber_ensure_pref_sanity
+
+/datum/unit_test/bubber_ensure_pref_sanity/Run()
+	for (var/datum/preference/preference as anything in GLOB.preference_entries)
+		preference = GLOB.preference_entries[preference]
+		if (preference.abstract_type == preference.type) // You're safe... for now.
+			continue
+
+		var/supplementals_failed = FALSE
+		var/list/data = preference.compile_constant_data()
+		if (data && !isnull(data[SUPPLEMENTAL_FEATURE_KEY]) && !islist(data[SUPPLEMENTAL_FEATURE_KEY]))
+			TEST_FAIL("[preference.type] : Supplemental feature list isn't actually a list!")
+			supplementals_failed = TRUE
+
+		if (istype(preference, /datum/preference/choiced/mutant))
+			mutant_checks(preference, supplementals_failed)
+
+		if (istype(preference, /datum/preference/mutant_color))
+			color_checks(preference, supplementals_failed)
+
+		if (istype(preference, /datum/preference/emissive_toggle))
+			emissive_checks(preference, supplementals_failed)
```

Comment:
Pretty sure this can be an else if since only one can pass at any given time
```suggestion
		if (istype(preference, /datum/preference/choiced/mutant))
			mutant_checks(preference, supplementals_failed)

		else if (istype(preference, /datum/preference/mutant_color))
			color_checks(preference, supplementals_failed)

		else if (istype(preference, /datum/preference/emissive_toggle))
			emissive_checks(preference, supplementals_failed)
```